packages/trackerless-network/src/logic/neighbor-discovery/NeighborUpdateRpcLocal.ts

```diff
@@ -25,11 +26,24 @@ export class NeighborUpdateRpcLocal implements INeighborUpdateRpc {
         this.config = config
     }
 
+    private createResponse(removeMe: boolean): NeighborUpdate {
+        return {
+            streamPartId: this.config.streamPartId,
+            neighborDescriptors: this.config.neighbors.getAll().map((neighbor) => neighbor.getPeerDescriptor()),
+            removeMe
+        }
+    }
+
     // INeighborUpdateRpc server method
     async neighborUpdate(message: NeighborUpdate, context: ServerCallContext): Promise<NeighborUpdate> {
         const senderPeerDescriptor = (context as DhtCallContext).incomingSourceDescriptor!
         const senderId = getNodeIdFromPeerDescriptor(senderPeerDescriptor)
-        if (this.config.neighbors.has(senderId)) {
+        if (this.config.neighbors.has(senderId) 
+            && this.config.neighbors.size() > this.config.neighborCount
+            && message.neighborDescriptors.length > this.config.neighborCount
```

Comment:
Here we compare the how many neighbors the remote has, and how many we'd like to have? But could it be that the remote has been configured differently, and its "preferered neighbor count" is something else than what is configured here in the local node?

Reply:
Does not matter. Then the remote node has to find someone with a free neighbor slot

Reply:
My intuition:

The local node always has a mandate to cut connections as necessary to arrive at its `neighborCount`config setting. This condition is represented by `this.config.neighbors.size() > this.config.neighborCount`.

The second condition `message.neighborDescriptors.length > this.config.neighborCount` is there to do it "politely". We only ask the other side to cut the connection if they are above a reasonable neighbor count (e.g.  the setting we are using).

Think of it like this: if our neighbor _A_ has only 2 neighbors of its own while another neighbor of ours, _B_, has 8 neighbors of its own, and we have too many neighbors right now,  which neighbor would you ask to cut connection with us?

Reply:
Maybe we could add a comment for line `43`. Something like this: we use `this.config.neighborCount` as an approximation (as we don't know the config value of used by the remote), and we kind of assume that if the remote has more neighbors than that, it has "enough neighbors" and therefore we can remove the connection.

Reply:
```
// We don't know the remote's neighborCount setting here. We only ask to cut connections if the remote has a "sufficient" number of neighbors, where "sufficient" means our neighborCount setting.
```